subsys/bluetooth/services/fast_pair/fp_crypto/fp_crypto_psa.c

```diff
@@ -0,0 +1,287 @@
+		LOG_ERR("Invalid psa_hash_compute output len: %zu", hash_len);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static psa_key_id_t import_hmac_sha256_key(const uint8_t *data, size_t len)
+{
+	psa_status_t status;
+	psa_key_id_t key_id = PSA_KEY_ID_NULL;
+	psa_key_attributes_t key_attr = PSA_KEY_ATTRIBUTES_INIT;
+
+	psa_set_key_usage_flags(&key_attr, PSA_KEY_USAGE_SIGN_HASH);
+	psa_set_key_lifetime(&key_attr, PSA_KEY_LIFETIME_VOLATILE);
+	psa_set_key_algorithm(&key_attr, PSA_ALG_HMAC(PSA_ALG_SHA_256));
+	psa_set_key_type(&key_attr, PSA_KEY_TYPE_HMAC);
+	psa_set_key_bits(&key_attr, len * CHAR_BIT);
+
+	status = psa_import_key(&key_attr, data, len, &key_id);
+	psa_reset_key_attributes(&key_attr);
+
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_import_key failed (err: %d)", status);
+		key_id = PSA_KEY_ID_NULL;
+	}
+
+	return key_id;
+}
+
+static int fp_crypto_psa_hmac_sha256(uint8_t *out, const uint8_t *in, size_t data_len,
+				     psa_key_id_t key_id)
+{
+	size_t olen = 0;
+	psa_status_t status = psa_mac_compute(key_id, PSA_ALG_HMAC(PSA_ALG_SHA_256), in, data_len,
+					      out, FP_CRYPTO_SHA256_HASH_LEN, &olen);
+
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_mac_compute failed (err: %d)", status);
+		return -EIO;
+	}
+
+	if (olen != FP_CRYPTO_SHA256_HASH_LEN) {
+		LOG_ERR("Invalid psa_mac_compute output length: %zu", olen);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+int fp_crypto_hmac_sha256(uint8_t *out, const uint8_t *in, size_t data_len, const uint8_t *hmac_key,
+			  size_t hmac_key_len)
+{
+	int err = 0;
+	psa_key_id_t hmac_key_id;
+	psa_status_t status;
+
+	hmac_key_id = import_hmac_sha256_key(hmac_key, hmac_key_len);
+	if (hmac_key_id == PSA_KEY_ID_NULL) {
+		LOG_ERR("import_hmac_sha256_key failed");
+		return -EIO;
+	}
+
+	err = fp_crypto_psa_hmac_sha256(out, in, data_len, hmac_key_id);
+
+	status = psa_destroy_key(hmac_key_id);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_destroy_key failed (err: %d)", status);
+		/* Overwrite error code to forward information about psa_destroy_key failure. */
+		err = -ECANCELED;
+	}
+
+	return err;
+}
+
+static psa_key_id_t import_aes128_key(const uint8_t *data)
+{
+	static const size_t len = FP_CRYPTO_AES128_KEY_LEN;
+
+	psa_status_t status;
+	psa_key_id_t key_id = PSA_KEY_ID_NULL;
+	psa_key_attributes_t key_attr = PSA_KEY_ATTRIBUTES_INIT;
+
+	psa_set_key_usage_flags(&key_attr, PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_DECRYPT);
+	psa_set_key_lifetime(&key_attr, PSA_KEY_LIFETIME_VOLATILE);
+	psa_set_key_algorithm(&key_attr, PSA_ALG_ECB_NO_PADDING);
+	psa_set_key_type(&key_attr, PSA_KEY_TYPE_AES);
+	psa_set_key_bits(&key_attr, len * CHAR_BIT);
+
+	status = psa_import_key(&key_attr, data, len, &key_id);
+	psa_reset_key_attributes(&key_attr);
+
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_import_key failed (err: %d)", status);
+		key_id = PSA_KEY_ID_NULL;
+	}
+
+	return key_id;
+}
+
+static int psa_aes128_ecb_crypt(uint8_t *out, const uint8_t *in, psa_key_id_t key_id, bool encrypt)
```

Comment:
maybe `fp_crypto` prefix should also be added here to not confuse it with PSA API like it was added in the `fp_crypto_psa_hmac_sha256`?

Reply:
addressed